pkg/services/sqlstore/sqlstore.go

```diff
@@ -150,11 +153,29 @@ func getEngine() (*xorm.Engine, error) {
 	}
 
 	sqlog.Info("Initializing DB", "dbtype", DbCfg.Type)
-	return xorm.NewEngine(DbCfg.Type, cnnstr)
+	engine, err := xorm.NewEngine(DbCfg.Type, cnnstr)
+	if err != nil {
+		return nil, err
+	} else {
+		if DbCfg.MaxConn != 0 {
```

Comment:
This if statement is kinda of redundant. xorm will ignore any connection setting that is 0. So just pass in the DbCfg.MaxConn value. Same for `MaxOpenConn` and `MaxIdleConn`